clients/rospy/src/rospy/core.py

```diff
@@ -142,20 +143,37 @@ def rospyerr(msg, *args):
 def rospywarn(msg, *args):
     """Internal rospy client library warn logging"""
     _rospy_logger.warn(msg, *args)
-    
-logdebug = logging.getLogger('rosout').debug
 
-logwarn = logging.getLogger('rosout').warning
+
+def _base_logger(msg, name=None, throttle=None, level='info'):
+
+    rospy_logger = logging.getLogger('rosout')
+    if name:
+        rospy_logger = rospy_logger.getChild(name)
+    logfunc = getattr(rospy_logger, level)
+
+    if throttle:
+        caller_id = _frame_record_to_caller_id(inspect.stack()[1])
+        _logging_throttle(caller_id, logfunc, throttle, msg)
+    else:
+        logfunc(msg)
+
+
+loginfo = partial(_base_logger, level='info')

```

Comment:
If we're only using `partial` for this one thing, it might be clearer to just have a getter function that returns a closure. Not really attached one way or another.

```
def _get_logger(level='info'):
    def _base_logger(msg, name=None, throttle=None):

        rospy_logger = logging.getLogger('rosout')
        if name:
            rospy_logger = rospy_logger.getChild(name)
        logfunc = getattr(rospy_logger, level)

        if throttle:
            caller_id = _frame_record_to_caller_id(inspect.stack()[1])
            _logging_throttle(caller_id, logfunc, throttle, msg)
        else:
            logfunc(msg)
    return _base_logger
```